appserver/payara-appserver-modules/microprofile/openapi/src/main/java/fish/payara/microprofile/openapi/impl/model/util/ModelUtils.java

```diff
@@ -552,30 +577,44 @@ public static String getResourcePath(GenericDeclaration declaration, Map<String,
         return null;
     }
 
-    private static String getResourcePath(Class<?> clazz, Map<String, Set<Class<?>>> resourceMapping) {
+    public static String getResourcePath(Type declaration, Map<String, Set<String>> resourceMapping) {
+        if (declaration instanceof MethodModel) {
+            return getResourcePath((MethodModel) declaration, resourceMapping);
+        } else if (declaration instanceof ClassModel) {
+            return getResourcePath((ClassModel) declaration, resourceMapping);
+        }
+        return null;
+    }
+
+    public static String getResourcePath(ClassModel clazz, Map<String, Set<String>> resourceMapping) {
         // If the class is a resource and contains a mapping
-        AnnotationInfo<?> annotations = AnnotationInfo.valueOf(clazz);
+        AnnotationInfo annotations = AnnotationInfo.valueOf(clazz);
         if (annotations.isAnnotationPresent(Path.class)) {
-            for (Map.Entry<String, Set<Class<?>>> entry : resourceMapping.entrySet()) {
-                if (entry.getValue() != null && entry.getValue().contains(clazz)) {
-                    return normaliseUrl(entry.getKey() + "/" + annotations.getAnnotation(Path.class).value());
+            for (Map.Entry<String, Set<String>> entry : resourceMapping.entrySet()) {
+                if (entry.getValue() != null && entry.getValue().contains(clazz.getName())) {
+                    return normaliseUrl(entry.getKey() + "/" + annotations.getAnnotationValue(Path.class));
                 }
             }
         }
         return null;
     }
 
-    private static String getResourcePath(Method method, Map<String, Set<Class<?>>> resourceMapping) {
-        AnnotationInfo<?> annotations = AnnotationInfo.valueOf(method.getDeclaringClass());
+    public static String getResourcePath(MethodModel method, Map<String, Set<String>> resourceMapping) {
```

Comment:
Moving from `Class` to `String` does take away a lot of the context. Maybe this should be using a wrapper type on `String` to become more typed again.